CorsixTH/Lua/world.lua

```diff
@@ -2072,45 +2072,11 @@ function World:objectPlaced(entity, id)
   end
 
   self.entities[#self.entities + 1] = entity
-  -- If it is a bench we're placing, notify queueing patients in the vicinity
-  if id == "bench" and entity.tile_x and entity.tile_y then
-    local notify_distance = 6
-    local w, h = self.map.th:size()
-    for tx = math.max(1, entity.tile_x - notify_distance), math.min(w, entity.tile_x + notify_distance) do
-      for ty = math.max(1, entity.tile_y - notify_distance), math.min(h, entity.tile_y + notify_distance) do
-        for _, patient in ipairs(self.entity_map:getHumanoidsAtCoordinate(tx, ty)) do
-          if class.is(patient, Patient) then
-            patient:notifyNewObject(id)
-          end
-        end
-      end
-    end
-  end
-  if id == "reception_desk" then
-    local numReceptionists = self.hospitals[1]:countStaffOfCategory("Receptionist")
-    if not self.ui.start_tutorial and numReceptionists == 0 then
-      -- TODO: Will not work correctly for multiplayer
-      self.ui.adviser:say(_A.room_requirements.reception_need_receptionist)
-    elseif numReceptionists > 0 and self.object_counts["reception_desk"] == 1 and
-        not self.hospitals[1].receptionist_msg and self.game_date:monthOfGame() > 3 then
-      self.ui.adviser:say(_A.warnings.no_desk_5)
-      self.hospitals[1].receptionist_msg = true
-    end
-  end
-  -- If it is a plant it might be advisable to hire a handyman
-  if id == "plant" and self.hospitals[1]:countStaffOfCategory("Handyman") == 0 then
-    self.ui.adviser:say(_A.staff_advice.need_handyman_plants)
-  end
-  if id == "gates_to_hell" then
-    entity:playEntitySounds("LAVA00*.WAV", {0,1350,1150,950,750,350},
-        {0,1450,1250,1050,850,450}, 40)
-    entity:setTimer(entity.world:getAnimLength(2550),
-                    --[[persistable:lava_hole_spawn_animation_end]]
-                    function(anim_entity)
-                      anim_entity:setAnimation(1602)
-                    end)
-    entity:setAnimation(2550)
-  end
+
+  -- Warn a hospital if that is possible.
+  if not entity.tile_x or not entity.tile_y then return end
+  local hosp = self:getHospital(entity.tile_x, entity.tile_y)
+  if hosp then hosp:objectPlaced(entity, id) end
```

Comment:
Likewise this would probably be world notifing all hospitals for multiplayer I . Again just getting ahead of myself and nothing is critical until that path is taken.

Reply:
Yes it's world in the central point, but that's how it currently is, it knows about every object and entity everywhere and keeps its own administration of it.

In multi-player however, the killer is not here but in the fact that all objects of a same-like kind etc are thrown together, so you'll have to iterate eg through all reception desks to find how many you got.
This also applies to staff, and perhaps patients. An array is simple, but performance of it is just horrible if you start with multiplayer.

Reply:
Indeed TH has all objects in a single array and traverses them 'all' when iterating for objects, however it also keeps indexed links between certain objects, like room to room object and room object to the next room object, and since all objects are the same size, everything is easily calculated. Again noting I don't expect a change.